specs/WebResourceRequested-CustomScheme.md

```diff
@@ -341,6 +345,40 @@ interface ICoreWebView2CustomSchemeRegistration : IUnknown {
   HRESULT SetAllowedOrigins(
     [in] UINT32 allowedOriginsCount,
     [in] LPCWSTR* allowedOrigins);
+
+  // Set this property to `true` if the URIs with this custom
+  // scheme will have an authority component (a host for custom schemes).
+  // Specifically, if you have a URI of the following form you should set the
+  // `HasAuthorityComponent` value as listed.
+  // | URI | Recommended HasAuthorityComponent value |
+  // | -- | -- |
+  // | ` custom-scheme-with-authority://host/path` | `true` |
+  // | `custom-scheme-without-authority:path` | `false` |
+  // When this property is set to `true`, the URIs with this scheme will be
+  // interpreted as having a
+  // [scheme and host](https://html.spec.whatwg.org/multipage/origin.html#concept-origin-tuple)
+  // origin similar to an http URI. Note that the port and user
+  // information are never included in the computation of origins for
+  // custom schemes.
+  // If this property is set to `false`, URIs with this scheme will have an
+  // [opaque origin](https://html.spec.whatwg.org/multipage/origin.html#concept-origin-opaque)
+  // similar to a data URI.
+  // This property is `false` by default.
+  //
+  // Note: For custom schemes registered as having authority component,
+  // navigations to URIs without authority of such custom schemes will fail.
+  // However, if the content inside WebView2 references
+  // a subresource with a URI that does not have
+  // an authority component, but of a custom scheme that is registered as
+  // having authority component, the URI will be interpreted as a relative path
+  // as specified in [RFC3986](https://www.rfc-editor.org/rfc/rfc3986).
+  // For example, custom-scheme-with-authority:path will be interpreted
+  // as custom-scheme-with-authority://host/path
+  // However, this behavior cannot be guaranteed to remain in future
+  // releases so it is recommended not to rely on this behavior.
```

Comment:
That's a really difficult restriction to leave to the documentation. Why not block it now, and only allow it in the future if we decide we can maintain it?

Reply:
With CoreWebView2.Navigate* we have a good place to do the check. For relative URIs the URI parsing is down in the depths of the web platform and afaik we don't have a good way to change behavior. This behavior does match http(s) URIs though so perhaps we could go easier on the wording here. Given that its how chromium is parsing http(s) URIs on the web I imagine its unlikely to change. @yildirimcagri can elaborate

Reply:
Like Dave said, the parsing of URIs with schemes but no host as relative urls in subresources is not easy to block, therefore I thought we should add documentation to describe the current behavior and warn the developer to (not) rely on it at their own risk. It is true that it is unlikely to change, should we decide to maintain it now?

Reply:
I will keep this as is for now. In future, we can decide to then maintain it. This is the unlikely case as well.